prompt2model/dataset_retriever/description_dataset_retriever.py

```diff
@@ -161,6 +165,37 @@ def canonicalize_dataset_using_columns_for_split(
             {"input_col": input_col, "output_col": output_col}
         )
 
+    @staticmethod
+    def automatic_column_selection(
+        instruction: str, dataset_name: str, dataset_columns: str, example_rows: dict
+    ) -> tuple[list[str], str]:
+        """Find appropriate input and output columns for a given dataset and tasks."""
+        prompt = construct_prompt_for_column_selection(
+            instruction, dataset_name, dataset_columns, example_rows
+        )
+        required_keys = ["input", "output"]
+        optional_keys = ["ambiguous", "irrelevant"]
+
+        response = parse_prompt_to_fields(prompt, required_keys, optional_keys)
+        input_columns = response["input"]
+        output_column = response["output"]
+
+        if len(input_columns) < 1 or len(output_column) != 1:
+            raise RuntimeError(
```

Comment:
Why is this RuntimeError, not a ValueError?

Reply:
I was thinking of it more as - hey if we don't have enough input/output columns then we can't proceed with this dataset, and it should be a runtime issue. Do you think that makes sense?

Reply:
According to the definition from StackOverflow: https://stackoverflow.com/questions/19763756/what-is-a-runtime-error

I agree that your point makes sense! Thanks for elaboration!